ui/src/app/settings/components/project-details/project-details.tsx

```diff
@@ -473,225 +435,560 @@ export class ProjectDetails extends React.Component<RouteComponentProps<{name: s
+                                            <div className='columns small-8'>
+                                                <FormField formApi={formApi} field={`spec.clusterResourceWhitelist[${i}].group`} component={AutocompleteField} />
+                                            </div>
+                                            <i
+                                                className='fa fa-times'
+                                                onClick={() => formApi.setValue('spec.clusterResourceWhitelist', removeEl(formApi.values.spec.clusterResourceWhitelist, i))}
+                                            />
+                                        </div>
+                                    ))}
+                                    <button
+                                        className='argo-button argo-button--short'
+                                        onClick={() =>
+                                            formApi.setValue(
+                                                'spec.clusterResourceWhitelist',
+                                                (formApi.values.spec.clusterResourceWhitelist || []).concat({
+                                                    group: '*',
+                                                    kind: '*'
+                                                })
+                                            )
+                                        }>
+                                        ADD RESOURCE
+                                    </button>
+                                </React.Fragment>
+                            )}
+                        </DataLoader>
+                    )}
+                    items={[]}
+                />
+                <EditablePanel
+                    save={item => this.saveProject(item)}
+                    values={proj}
+                    title={<React.Fragment>CLUSTER RESOURCE DENY LIST {helpTip('Cluster-scoped K8s API Groups and Kinds which are not permitted to be deployed')}</React.Fragment>}
+                    view={
+                        <React.Fragment>
+                            {proj.spec.clusterResourceBlacklist
+                                ? proj.spec.clusterResourceBlacklist.map((resource, i) => (
+                                      <div className='row white-box__details-row' key={i}>
+                                          <div className='columns small-4'>{resource.kind}</div>
+                                          <div className='columns small-8'>{resource.group}</div>
+                                      </div>
+                                  ))
+                                : emptyMessage('cluster resource deny list')}
+                        </React.Fragment>
+                    }
+                    edit={formApi => (
+                        <DataLoader load={() => services.clusters.list()}>
+                            {clusters => (
```

Comment:
Looks like `clusters` is not used here. Please remove `<DataLoader load={() => services.clusters.list()}>`